### PR TITLE
Improve workaround

### PR DESCRIPTION
**Pull request by martincostello**
Improve warning workarounds for native AoT.

### CODE REVIEW

**Comment by martincostello** [comment]
Need to check if this is a .NET or this-repo issue:

```
ILC : Trim analysis warning IL2026: __OptionValidationStaticInstances.<Validators_g>F8B1F3D42962A35D8FF1B1489612AEF6C36F3713335EFF79DB68A25973333C495____Attributes..cctor(): Using member 'System.ComponentModel.DataAnnotations.RangeAttribute.RangeAttribute(Type,String,String)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Generic TypeConverters may require the generic types to be annotated. For example, NullableConverter requires the underlying type to be DynamicallyAccessedMembers All. [/home/runner/work/dotnet-8-samples/dotnet-8-samples/src/WeatherApi/WeatherApi.csproj]
ILC : Trim analysis warning IL2026: __OptionValidationStaticInstances.<Validators_g>F8B1F3D42962A35D8FF1B1489612AEF6C36F3713335EFF79DB68A25973333C495____Attributes..cctor(): Using member 'System.ComponentModel.DataAnnotations.RangeAttribute.RangeAttribute(Type,String,String)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Generic TypeConverters may require the generic types to be annotated. For example, NullableConverter requires the underlying type to be DynamicallyAccessedMembers All. [/home/runner/work/dotnet-8-samples/dotnet-8-samples/src/WeatherApi/WeatherApi.csproj]
ILC : Trim analysis warning IL2026: __OptionValidationStaticInstances.<Validators_g>F8B1F3D42962A35D8FF1B1489612AEF6C36F3713335EFF79DB68A25973333C495____Attributes..cctor(): Using member 'System.ComponentModel.DataAnnotations.RangeAttribute.RangeAttribute(Type,String,String)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Generic TypeConverters may require the generic types to be annotated. For example, NullableConverter requires the underlying type to be DynamicallyAccessedMembers All. [/home/runner/work/dotnet-8-samples/dotnet-8-samples/src/WeatherApi/WeatherApi.csproj]
```

**Comment by martincostello** [comment]
Warnings appear to be https://github.com/dotnet/extensions/issues/4622.

**Comment by eerhardt** [comment]
> Need to check if this is a .NET or this-repo issue:
> 
> ```
> ILC : Trim analysis warning IL2026: __OptionValidationStaticInstances.<Validators_g>F8B1F3D42962A35D8FF1B1489612AEF6C36F3713335EFF79DB68A25973333C495____Attributes..cctor(): Using member 'System.ComponentModel.DataAnnotations.RangeAttribute.RangeAttribute(Type,String,String)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Generic TypeConverters may require the generic types to be annotated. For example, NullableConverter requires the underlying type to be DynamicallyAccessedMembers All. [/home/runner/work/dotnet-8-samples/dotnet-8-samples/src/WeatherApi/WeatherApi.csproj]
> ILC : Trim analysis warning IL2026: __OptionValidationStaticInstances.<Validators_g>F8B1F3D42962A35D8FF1B1489612AEF6C36F3713335EFF79DB68A25973333C495____Attributes..cctor(): Using member 'System.ComponentModel.DataAnnotations.RangeAttribute.RangeAttribute(Type,String,String)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Generic TypeConverters may require the generic types to be annotated. For example, NullableConverter requires the underlying type to be DynamicallyAccessedMembers All. [/home/runner/work/dotnet-8-samples/dotnet-8-samples/src/WeatherApi/WeatherApi.csproj]
> ILC : Trim analysis warning IL2026: __OptionValidationStaticInstances.<Validators_g>F8B1F3D42962A35D8FF1B1489612AEF6C36F3713335EFF79DB68A25973333C495____Attributes..cctor(): Using member 'System.ComponentModel.DataAnnotations.RangeAttribute.RangeAttribute(Type,String,String)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Generic TypeConverters may require the generic types to be annotated. For example, NullableConverter requires the underlying type to be DynamicallyAccessedMembers All. [/home/runner/work/dotnet-8-samples/dotnet-8-samples/src/WeatherApi/WeatherApi.csproj]
> ```

@martincostello  - that issue will resolve those warnings, but it will take a couple servicing releases. The root underlying issue is https://github.com/dotnet/runtime/issues/94119, which will hopefully be fixed in the `8.0.2` servicing release of dotnet/runtime.

File: src/WeatherApi/WeatherApi.csproj [hunk 1]
```diff
@@ -11,8 +11,6 @@
   <!-- Add support for Ahead-of-Time (AoT) compilation -->
   <PropertyGroup Condition="!$([System.OperatingSystem]::IsMacOS())">
```
Comment (eerhardt) [inline src/WeatherApi/WeatherApi.csproj:12]: macOS is supported in NativeAOT in .NET 8. So this condition can be removed if you want.

File: src/WeatherApi/WeatherApi.csproj [hunk 2]
```diff
@@ -11,8 +11,6 @@
...
     <InvariantGlobalization>true</InvariantGlobalization>
-    <!-- HACK Suppress IL warnings from Microsoft.Extensions.Http.Resilience until 8.0.0 -->
```
Comment (eerhardt) [inline src/WeatherApi/WeatherApi.csproj:14]: If you want to disable warnings coming from a specific set of libraries, a strategy I found that works is to use `<TrimmerSingleWarn>false</TrimmerSingleWarn>` and disable the warning for `IL2104` (the warning number for a single warning for a whole library).

And then turn on "single warnings" for the specific libraries you want to suppress. You can see an example of that here:
https://github.com/dotnet/eShop/blob/1e803b09ba8caa3702f037deac837fab956abe33/Directory.Build.targets#L7-L23

File: src/WeatherApi/WeatherApi.csproj (hunk 1 above)
Comment (martincostello) [reply src/WeatherApi/WeatherApi.csproj]: So it works here, but it _doesn't_ work in Polly 🤔 ([logs](https://github.com/App-vNext/Polly/actions/runs/6880981184/job/18716366049?pr=1800#step:5:44))

```
Error: /Users/runner/.dotnet/sdk/8.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkRefe enceResolution.targets(90,5): error NETSDK1204: Ahead-of-time compilation is not supported on the current platform 'osx-x64'. [/Users/runner/work/Polly/Polly/test/Polly.AotTest/Polly.AotTest.csproj]
```

Looks like it's during `dotnet clean` (which this repo doesn't do) so maybe that's a bug somewhere?

Comment (martincostello) [reply src/WeatherApi/WeatherApi.csproj]: FYI: https://github.com/dotnet/sdk/issues/36960

**Pull request merged**